### ISSUE TITLE
Expressing anchor operator in Emacs Lisp

### ISSUE THREAD

**Issue by sogaiu** [ubolonton/emacs-tree-sitter#99]
TLDR:

It appears that using the anchor operator (dot) as part of expressing a query in Emacs Lisp (not as a string) can be problematic.

---

Longer version:

I tried expressing a query using Emacs Lisp in a manner similar to what is done here: https://github.com/emacs-csharp/csharp-mode/blob/98a179ad9c8bb7c816986cdd5b9afbe2fcc77f78/csharp-tree-sitter.el#L43-L227

That starts off like:
```
(defvar-local csharp-mode-tree-sitter-patterns
  [ ;; Various constructs
   (comment) @comment
   (modifier) @keyword
   (this_expression) @keyword
```

Note that the query is not expressed as as string, but rather a non-string form.

It happens that for the query I'm trying there is a section like:
```
   (tuple
    .
    (symbol) @function
    (.match? @function "^(\\*|\\+|\\-|/|<|<=|=|>|>=|_|\\.break|\\.breakall|\\.bytecode|\\.clear|\\.clearall|\\.disasm|\\.fiber|\\.fn|\\.frame|...
```
With that though, Emacs gives the following error at start up:
```
Invalid read syntax: . in wrong context
```
Note that the query is using the [anchor operator](
https://tree-sitter.github.io/tree-sitter/using-parsers#anchors) which is expressed using a dot.

My current guess is that if the anchor operator is expressed using a single dot, this doesn't work so well with Emacs' dotted pair notation.

Expressing the query as a string works fine, so that appears to one work-around.

Is this a known issue?

**Comment by ubolonton** [comment]
Yes, it's a known issue that hasn't been documented.

The suggested workaround is putting the query patterns in a file, instead of embedding it in Emacs Lisp.

**Comment by sogaiu** [comment]
Thank you!

I will do so.

**Comment by ymarco** [comment]
Maybe `\.` would work?
```emacs-lisp
(symbol-name (nth 1 '(a \. b c)))
  => "."
```

**Comment by sogaiu** [comment]
@ymarco Thanks for the idea.

I gave that a try and it seems to be working so far :)
